pkg/storegateway/bucket_store_metrics.go

```diff
@@ -212,7 +212,7 @@ func (m *BucketStoreMetrics) Collect(out chan<- prometheus.Metric) {
 	data.SendSumOfCounters(out, m.blockDrops, "thanos_bucket_store_block_drops_total")
 	data.SendSumOfCounters(out, m.blockDropFailures, "thanos_bucket_store_block_drop_failures_total")
 
-	data.SendSumOfGauges(out, m.blocksLoaded, "thanos_bucket_store_blocks_loaded")
+	data.SendSumOfGaugesPerUser(out, m.blocksLoaded, "thanos_bucket_store_blocks_loaded")
```

Comment:
Will this cause an explosion in cardinality if there is a high churn in usera?

Reply:
My 2 cents:
1. We have per user metrics already in other Cortex components
2. For cardinality, I feel user label is okay. Compared to short lived job names like pod name, container ID, etc, user ID should be relatively bounded.

Reply:
I think this change is ok, but it looks like store-gateway only "soft deletes" per user metrics: https://github.com/cortexproject/cortex/blob/978b35c6afb874bb7284d90f348e986ca7622570/pkg/storegateway/bucket_store_metrics.go#L171

If that's the case, turning a metric to per-user may not be a good idea because of memory leak.

Reply:
actually, I think this PR would prevent memory leak https://github.com/cortexproject/cortex/pull/4813

so, I think we are good to go.

Reply:
If you are talking about benchmarking and continuously tests then every label can be "short lived", right? If we are not reusing the same test user.

Reply:
yea :)